data-serving/scripts/converters.py

```diff
@@ -1,209 +1,267 @@
+
+    Returns:
+      None: When the input value is empty.
+      Dict[str, Any]: When the value is present and successfully parsed. The
+        dictionary is in the format:
+        {
+          'start': Any,
+          'end': Any
+        }
+        where start == end if value represents a single value, as opposed to a
+        range, and where the type of the values is dictated by the parser and
+        formatter.
     '''
-    if pd.isna(value):
-        return None
+    start, end = parse_range(value, parser)
 
-    # First check if the value is represented as a range.
-    if type(value) is str and value.find('-') >= 0:
-        value_range = trim_string_array(value.split('-'))
+    range = {}
+    if start is not None:
+        range['start'] = formatter(start)
+    if end is not None:
+        range['end'] = formatter(end)
 
-        # Handle open ranges (i.e. missing min or max).
-        first_val = parse_fn(value_range[0])
-        if len(value_range) == 1:
-            return {
-                'end': format_fn(first_val)
+    return range if range else None
 
-            } if value.startswith('-') else {
-                'start': format_fn(first_val)
-            }
-
-        # Handle cases with a min and max.
-        second_val = parse_fn(value_range[1])
-        return {
-            'start': format_fn(first_val),
-            'end': format_fn(second_val)
 
+def convert_age_range(ages: Any) -> Dict[str, float]:
+    '''
+    Converts an age value to the expected age range output format. This wraps
+    convert_range with a date-specific parser and formatter.
+
+    Parameters:
+      ages: A representation of an age expected to be a string in one of the
+        following formats:
+        - "x": Single age in years.
+        - "x months": Single age in months.
+        - "x weeks": Single age in weeks.
+        - "x - y": Age range.
+        - "x -": Age range start.
+        - "- y": Age range end.
+
+    Raises:
+      ValueError if the value can't be successfully parsed into an age.
+    Returns:
+      None: When the input value is empty.
+      Dict[str, float]: When the value is present and successfully parsed. The
+        dictionary in the format:
+        {
+          'start': float,
+          'end': float
         }
-
-    # We have a specific value. We create a range with identical min and max
-    # values.
-    parsed_value = parse_fn(value)
-    return {
-        'start': format_fn(parsed_value),
-        'end': format_fn(parsed_value)
-    }
+        where start == end if value represents a single age.
+    '''
+    return convert_range(ages, parse_age, lambda x: x)
 
 
-def convert_date_range(date_str: str) -> Dict[str, Any]:
-    return convert_range(
-        date_str,
-        lambda x: datetime.datetime.strptime(x, '%d.%m.%Y'),
-        lambda x: {
-            '$date': x.strftime('%Y-%m-%dT%H:%M:%SZ')
-        })
+def convert_date_range(dates: Any) -> Dict[str, Dict[str, str]]:
```

Comment:
nice! I generally left the input types as Any since it could be anything from the CSV (string, float, int, bool) -- does that make sense or should I be typing for the values I generally expect and handle (or typing all inputs as Union[float, str, bool...]?